File: leetcode/150.evaluate-reverse-polish-notation/cpp/evalrpn_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "evalrpn.cpp"

struct testCase
{
    std::vector<std::string> tokens;
    int want;
};

TEST_CASE("Eval RPN")
{
    std::vector<testCase>
        tcs{
            {{"2", "1", "+", "3", "*"}, 9},
            {{"4", "13", "5", "/", "+"}, 6},
            {{"10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"}, 22},
        };

    Solution s;

    for (auto tc : tcs)
    {
        auto got = s.evalRPN(tc.tokens);
        CHECK(got == tc.want);
    }
}

File: leetcode/150.evaluate-reverse-polish-notation/go/evalrpn_test.go
package evalrpn

import "testing"

type testcase struct {
	tokens []string
	want   int
}

func TestEvalRpn(t *testing.T) {
	tcs := []testcase{
		{tokens: []string{"2", "1", "+", "3", "*"}, want: 9},
		{tokens: []string{"4", "13", "5", "/", "+"}, want: 6},
		{tokens: []string{"10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"}, want: 22},
	}

	for _, tc := range tcs {
		got := evalRPN(tc.tokens)
		if got != tc.want {
			t.Fatalf("evalrpn, want %d, got %d", tc.want, got)
		}
	}
}
